### PR TITLE
3284 Normalize Provider name for improving search results

### PR DESCRIPTION
**Pull request by aliuk2012**
### Context
#1310 will be split into three smaller PR and deployed individually. This is the two of three.

Updates the data so that we can use the new column for improved provider search results 
### Changes proposed in this pull request
- Adds a new column `provider_name_search` to Provider table
- Migration to populate the new column
- Adds a "not null" constraint to the new column
- Adds a "not null" constraint toDB and model validation to Provider
### Checklist

- [ ] Make sure all information from the Trello card is in here
- [ ] Attach to Trello card
- [ ] Rebased master
- [ ] Cleaned commit history
- [ ] Tested by running locally

### CODE REVIEW

File: app/models/provider.rb [hunk 1]
```diff
@@ -310,4 +314,8 @@ def services
       )
     end
   end
+
+  def set_provider_name_search
+    self.provider_name_search = QueryNormalizerService.call(query: self.provider_name)
```
Comment (defong) [inline app/models/provider.rb:319]: does it need a token test... ie, mock was called etc?

File: spec/models/provider_spec.rb [hunk 2]
```diff
@@ -390,6 +400,19 @@
     end
   end
 
+  describe "before_save" do
+    subject { create :provider, provider_name: "Location 1 (Removed spaces and special characters)"}
```
Comment (defong) [inline spec/models/provider_spec.rb:404]: linter

File: app/models/provider.rb (hunk 1 above)
Comment (gpeng) [reply app/models/provider.rb:319]: I agree... I think we might be better with a mocked test here as the `QueryNormalizerService` is tested already. We just need to make sure it is used.

**Comment by gpeng** [comment]
Closing in favour of https://github.com/DFE-Digital/teacher-training-api/pull/1332

**Pull request closed without merging**